### DISCUSSION TITLE
Can't run TCP and UDP on Duplicate port

### DISCUSSION THREAD

**Discussion by jsleep** [wolveix/satisfactory-server#329] [Q&A] [answered]
I'm a weirdo trying to set up a project to run multiple different game servers all configured easily to run on Azure Container Instance.  ACI of course doesn't allow me to run TCP/UDP protocol on the same network port. Is there any way I can configure the image to run TCP/UDP on different ports?

**Comment by saedrin** [comment]
In the container configuration you should be able to change the port mapping here as an example splitting the ports between protocols:    

```
ports:
      - '7777:7777/udp'
      - '7778:7777/tcp'
```

**Reply by wolveix** [reply]
This would work in other cases, I don't believe it'll work for Satisfactory as the game (UDP) notifies the client of the API (TCP) port. If you're redirecting it via Docker, it breaks this, causing the API connection failure

**Comment by jsleep** [comment]
ACI does not support port forwarding - I'm thinking of just pivoting to VMs and running docker[compose] directly on those instead, so I should have more control over the network port rules - but just for knowledge before that, it would be helpful if I could potentially just configure these to run on different ports.

**Comment by gotanks0407** [comment]
If you are running different instances of the server at the same time you will have to run them all at different ports. Server 1 uses 7777, server 2 uses 7778, server 3 uses 7779. Or you need to have different incoming interfaces/public IPs for each to route too separately. 

But to your original point if you are bridging them all to the same port on different instances, no you cant do that. or traffic from server 1 could go to server 2. you would have no way to differentiate the traffic to each server. that is why you need different ports.

**Comment by jsleep** [comment]
I'm only running 1 server/instance - this is just for setting up any number of different game servers for different games, but anyone that uses it should only be using 1 server at a time per game. My issue is that Azure Container instances does not allow for port forwarding and reusing the same port for different protocols. I'm going to just configure a VM to do this I guess

**Comment by wolveix** [comment, marked as answer]
@jsleep no, unfortunately the game server does not allow you to specify different ports. I'm hoping Coffee Stain will include that functionality soon, but for now you're out of luck

**Reply by jsleep** [reply]
thanks for the help everyone, VM is now running with it, not as straightforward to configure unfortunately and won't be an easy automated deployment but it does run
